From 8905f7207866d4f7030036f51cb2e1aac5faaf95 Mon Sep 17 00:00:00 2001
From: HungTran <hung.tran.jy@rvc.renesas.com>
Date: Sat, 6 Feb 2016 00:46:45 +0700
Subject: [PATCH] Update for Flexible resolution

This patch updates many points:
- Revise scale code (easier to read now)
- Scale button also (last implement cannot scale button)
- Adjust position of some modules, such as CAM2, 3, 4 titles ...
- Reduce size of Mike icon for suitable with other

Signed-off-by: HungTran <hung.tran.jy@rvc.renesas.com>
---
 qml/simpleplayer/main.qml | 590 ++++++++++++++++++++++++++--------------------
 1 file changed, 333 insertions(+), 257 deletions(-)

diff --git a/qml/simpleplayer/main.qml b/qml/simpleplayer/main.qml
index 64a506d..0fe7981 100755
--- a/qml/simpleplayer/main.qml
+++ b/qml/simpleplayer/main.qml
@@ -794,16 +794,17 @@ Rectangle {
         width: container.width
         height: container.height
         color: "transparent"
-
+        property double scalew: (width / 1280)    // Default width is 1280
+        property double scaleh: (height / 720)    // Default height is 720
 
         /* change resolution button */
         Rectangle {
             id: btn_show
-            x: (root.width)/(1280/10) + micro.width + (root.width)/(1280/20)
-            y: (root.height)/(720/480)
+            x: (10 * root.scalew) + micro.width + (20 * root.scalew)
+            y: 480 * root.scaleh
             z: 10
-            width: 60
-            height: 60
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
             color: "transparent"
             Image {
                 id: screen_icon
@@ -829,8 +830,8 @@ Rectangle {
             id: btn_4_3
             x: btn_show.x
             y: btn_show.y
-            width: 60
-            height: 60
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
             color: "transparent"
             z: 5
 
@@ -875,8 +876,8 @@ Rectangle {
             id: btn_5_4
             x: btn_show.x
             y: btn_show.y
-            width: 60
-            height: 60
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
             color: "transparent"
             z: 5
             Image {
@@ -919,8 +920,8 @@ Rectangle {
             id: btn_16_10
             x: btn_show.x
             y: btn_show.y
-            width: 60
-            height: 60
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
             color: "transparent"
             z: 5
             Image {
@@ -963,8 +964,8 @@ Rectangle {
             id: btn_full_screen
             x: btn_show.x
             y: btn_show.y
-            width: 60
-            height: 60
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
             color: "transparent"
             z: 5
             enabled: false
@@ -1064,11 +1065,15 @@ Rectangle {
             id: title_display1
             color: "#00000000"
             visible: false
-            x: root.width - title_image1.width - (root.width)/(1280/320) - (root.width)/(1280/20)
-            y: (root.height)/(720/10)
+            x: root.width - title_image1.width - (340  * root.scalew)
+            y: 10 * root.scaleh
             Image {
                 id: title_image1
                 source: "pics/Title_Cam1.png"
+
+                // Original size is 100x42 pixel
+                width:  100 * root.scalew
+                height: 42 * root.scaleh
             }
         }
 
@@ -1077,10 +1082,15 @@ Rectangle {
             id: title_display2
             color: "#00000000"
             visible: false
-            x: title_display1.x + (root.width)/(1280/110); y: title_display1.y
+            x: title_display1.x + (130  * root.scalew)
+            y: title_display1.y
             Image {
                 id: title_image2
                 source: "pics/Title_Cam2.png"
+
+                // Original size is 100x42 pixel
+                width:  100 * root.scalew
+                height: 42 * root.scaleh
             }
         }
 
@@ -1089,10 +1099,15 @@ Rectangle {
             id: title_display3
             color: "#00000000"
             visible: false
-            x: title_display2.x; y: title_display2.y + (root.height)/(720/236)
+            x: title_display2.x
+            y: title_display2.y + (236 * root.scaleh)
             Image {
                 id: title_image3
                 source: "pics/Title_Cam3.png"
+
+                // Original size is 100x42 pixel
+                width:  100 * root.scalew
+                height: 42 * root.scaleh
             }
         }
 
@@ -1101,10 +1116,15 @@ Rectangle {
             id: title_display4
             color: "#00000000"
             visible: false
-            x: title_display3.x; y: title_display3.y + (root.height)/(720/237)
+            x: title_display3.x
+            y: title_display3.y + (237 * root.scaleh)
             Image {
                 id: title_image4
                 source: "pics/Title_Cam4.png"
+
+                // Original size is 100x42 pixel
+                width:  100 * root.scalew
+                height: 42 * root.scaleh
             }
         }
 
@@ -1121,13 +1141,14 @@ Rectangle {
             }
         }
 
-        /*Micro controller*/
+        /*Mike controller*/
         Rectangle {
             id: micro
             color: "transparent"
-            width: 60; height: 60
-            x: (root.width)/(1280/10)
-            y: (root.height)/(720/480)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 10 * root.scalew
+            y: 480 * root.scaleh
             z: 1
             Image{
                 id: micro_image
@@ -1181,9 +1202,10 @@ Rectangle {
         Rectangle {
             id: arrow
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/10)
-            y: (root.height)/(720/480)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 10 * root.scalew
+            y: 480 * root.scaleh
             visible: false
             Image{
                 id: arrow_image
@@ -1252,13 +1274,15 @@ Rectangle {
         Rectangle {
             id: play
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/15)
-            y: (root.height)/(720/180)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 15 * root.scalew
+            y: 180 * root.scaleh
             visible: false
             Image{
                 id: play_image
-                width: 50; height: 50
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
                 source: ""
             }
             MouseArea {
@@ -1318,13 +1342,15 @@ Rectangle {
         Rectangle {
             id: stop
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/15)
-            y: (root.height)/(720/230)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 15 * root.scalew
+            y: 230 * root.scaleh
             visible: false
             Image{
                 id: stop_image
-                width: 50; height: 50
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
                 source: ""
             }
             MouseArea {
@@ -1365,13 +1391,16 @@ Rectangle {
         Rectangle {
             id: rec
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/15)
-            y: (root.height)/(720/280)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 15 * root.scalew
+            y: 280 * root.scaleh
             visible: false
             Image{
                 id: rec_image
                 source: ""
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
             }
             MouseArea {
                 anchors.fill: parent
@@ -1424,13 +1453,15 @@ Rectangle {
         Rectangle {
             id: rec1
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/70)
-            y: (root.height)/(720/205)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 70 * root.scalew
+            y: 205 * root.scaleh
             visible: false
             Image{
                 id: rec1_image
-                width: 50; height: 50
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
                 source: ""
             }
             MouseArea {
@@ -1494,13 +1525,15 @@ Rectangle {
         Rectangle {
             id: rec2
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/70)
-            y: (root.height)/(720/255)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 70 * root.scalew
+            y: 255 * root.scaleh
             visible: false
             Image{
                 id: rec2_image
-                width: 50; height: 50
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
                 source: ""
             }
             MouseArea {
@@ -1564,13 +1597,15 @@ Rectangle {
         Rectangle {
             id: rec3
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/70)
-            y: (root.height)/(720/305)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 70 * root.scalew
+            y: 305 * root.scaleh
             visible: false
             Image{
                 id: rec3_image
-                width: 50; height: 50
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
                 source: ""
             }
             MouseArea {
@@ -1634,13 +1669,15 @@ Rectangle {
         Rectangle {
             id: rec4
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/70)
-            y: (root.height)/(720/355)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 70 * root.scalew
+            y: 355 * root.scaleh
             visible: false
             Image{
                 id: rec4_image
-                width: 50; height: 50
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
                 source: ""
             }
             MouseArea {
@@ -1704,33 +1741,36 @@ Rectangle {
         Rectangle {
             id: view
             color: "transparent"
-            width: 50; height: 50
-            x:(root.width)/(1280/30)
-            y:(root.height)/(720/120)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 30 * root.scalew
+            y: 120 * root.scaleh
             z:1
             visible: false
             property var mode: 1
             Image{
                 id: view_image
                 source: list_icon_red_01.get(2).icon_normal
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
             }
             MouseArea {
                 anchors.fill: parent
                 hoverEnabled: true
                 onClicked: {
-                    if (view.mode == 1) {
+                    if (view.mode == 1) {   // Normal menu view
                         view.mode = 2
                         menu.visible = false
                         temp_status.visible = false
                         micro.visible = false
-                        micro.x = (root.width)/(1280/15)
-                        micro.y = (root.height)/(720/430)
+                        micro.x = 15 * root.scalew
+                        micro.y = 430 * root.scaleh
                         view.visible = false
-                        view.x = (root.width)/(1280/70)
-                        view.y = (root.height)/(720/305)
+                        view.x = 70 * root.scalew
+                        view.y = 305 * root.scaleh
                         mail.visible = false
-                        mail.x = (root.width)/(1280/70)
-                        mail.y = (root.height)/(720/355)
+                        mail.x = 70 * root.scalew
+                        mail.y = 355 * root.scaleh
                         arrow_state = arrow_state_normal
                         arrow_image.source = image_arrow
                         arrow.visible = true
@@ -1759,32 +1799,32 @@ Rectangle {
                         set_to_active_all_list()
                         set_state = set_state_normal
                     }
-                    else if (view.mode == 2) {
+                    else if (view.mode == 2) {   // Compact menu (hidden bar at the left)
                         view.mode = 1
                         menu.visible = true
                         pathv.visible = true
                         temp_status.visible = true
                         micro.visible = true
-                        micro.x = (root.width)/(1280/10)
-                        micro.y = (root.height)/(720/480)
-                        view.x = (root.width)/(1280/30)
-                        view.y = (root.height)/(720/120)
-                        mail.x = (root.width)/(1280/80)
-                        mail.y = (root.height)/(720/120)
+                        micro.x = 10 * root.scalew
+                        micro.y = 500 * root.scaleh
+                        view.x = 30 * root.scalew
+                        view.y = 120 * root.scaleh
+                        mail.x = 80 * root.scalew
+                        mail.y = 120 * root.scaleh
                         set_to_hide_ball_list()
                         arrow.visible = false
                         set_state = set_state_normal
                         if (display2_state == camera1_state) {
-                            micro.x = (root.width)/(1280/970)
-                            micro.y = (root.height)/(720/180)
+                            micro.x = 970 * root.scalew
+                            micro.y = 180 * root.scaleh
                         }
                         if (display3_state == camera1_state) {
-                            micro.x = (root.width)/(1280/970)
-                            micro.y = (root.height)/(720/420)
+                            micro.x = 970 * root.scalew
+                            micro.y = 420 * root.scaleh
                         }
                         if (display4_state == camera1_state) {
-                            micro.x = (root.width)/(1280/970)
-                            micro.y = (root.height)/(720/655)
+                            micro.x = 970 * root.scalew
+                            micro.y = 655 * root.scaleh
                         }
                         set_ball_normal_list_01(list_state,view_image,2)
                         switch (list_state) {
@@ -1845,14 +1885,17 @@ Rectangle {
         Rectangle {
             id: mail
             color: "transparent"
-            width: 50; height: 50
-            x:(root.width)/(1280/80)
-            y:(root.height)/(720/120)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 80 * root.scalew
+            y: 120 * root.scaleh
             z:1
             visible: false
             Image{
                 id: mail_image
                 source: list_icon_red_01.get(4).icon_normal
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
             }
             MouseArea {
                 anchors.fill: parent
@@ -1895,13 +1938,16 @@ Rectangle {
         Rectangle {
             id: set
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/15)
-            y: (root.height)/(720/330)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 15 * root.scalew
+            y: 330 * root.scaleh
             visible: false
             Image{
                 id: set_image
                 source: ""
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
             }
             MouseArea {
                 anchors.fill: parent
@@ -1951,13 +1997,15 @@ Rectangle {
         Rectangle {
             id: color
             color: "transparent"
-            width: 50; height: 50
-            x: (root.width)/(1280/15)
-            y: (root.height)/(720/380)
+            width: 50 * root.scalew
+            height: 50 * root.scaleh
+            x: 15 * root.scalew
+            y: 380 * root.scaleh
             visible: false
             Image{
                 id: color_image
-                width: 50; height: 50
+                width: 50 * root.scalew
+                height: 50 * root.scaleh
                 source: ""
             }
             MouseArea {
@@ -2036,9 +2084,10 @@ Rectangle {
         Rectangle {
             id: temp_status
             color: "transparent"
-            width: 90; height: 90
-            x: (root.width)/(1280/25)
-            y: (root.height)/(720/10)
+            width: 90 * root.scalew
+            height: 90 * root.scaleh
+            x: 25 * root.scalew
+            y: 10 * root.scaleh
             z:9
             visible: true
             MouseArea {
@@ -2079,26 +2128,31 @@ Rectangle {
         Rectangle {
             id: menu
             color: "transparent"
-            width: 160; height: 160
-            x: (root.width)/(1280/10)
-            y: (root.height)/(720/10)
+            width: 160 * root.scalew
+            height: 160 * root.scaleh
+            x: 10 * root.scalew
+            y: 10 * root.scaleh
             visible: true
             //Status ball
             Image{
                 id: status_image
-                x: 15
-                y: 10
+                x: 15 * root.scalew
+                y: 10 * root.scaleh
+                width: 120 * root.scalew
+                height: 120 * root.scaleh
                 source: status_ball
             }
 
             //Status text
             Rectangle {
                 id: status_text
-                x: 23
-                y: 25
+                x: 23 * root.scalew
+                y: 25 * root.scaleh
                 Image{
                     x: 0
-                    y: 20
+                    y: 20 * root.scaleh
+                    width:  100 * root.scalew
+                    height: 30  * root.scaleh
                     anchors.centerIn: status_text.Center
                     id: status_text1
                     source: text_live_red
@@ -2106,7 +2160,9 @@ Rectangle {
 
                 Image{
                     x: 0
-                    y: 40
+                    y: 40 * root.scaleh
+                    width:  90 * root.scalew
+                    height: 20  * root.scaleh
                     anchors.centerIn: status_text.Center
                     id: status_text2
                     source: ""
@@ -2120,26 +2176,26 @@ Rectangle {
                 delegate: Component {
                     id: delegate
                     Item {
-                        width: 50; height: 50
+                        width: 60 * root.scalew
+                        height: 60 * root.scaleh
                         scale: PathView.iconScale
                         opacity: PathView.iconOpacity
-                        Column {
-                            Rectangle{
-                                id: icon_frame
-                                width: 64; height: 64
-                                color: "#00000000"
-                                Image {
-                                    id: icon_image
-                                    anchors.verticalCenter: icon_frame.verticalCenter
-                                    anchors.horizontalCenter: icon_frame.horizontalCenter
-                                    //anchors.horizontalCenter: nameText.horizontalCenter
-                                    width: 64
-                                    height: 64
-                                    source: icon
-                                }
-                                //Text { id: nameText; text: name; font.pointSize: 16 }
-
+                        Rectangle{
+                            id: icon_frame
+                            width: 60 * root.scalew
+                            height: 60 * root.scaleh
+                            color: "#00000000"
+                            Image {
+                                id: icon_image
+                                anchors.verticalCenter: icon_frame.verticalCenter
+                                anchors.horizontalCenter: icon_frame.horizontalCenter
+                                //anchors.horizontalCenter: nameText.horizontalCenter
+                                width: 64 * root.scalew
+                                height: 64 * root.scaleh
+                                source: icon
                             }
+                            //Text { id: nameText; text: name; font.pointSize: 16 }
+
                         }
                         MouseArea {
                             anchors.fill: parent
@@ -2157,18 +2213,18 @@ Rectangle {
 
                             onPressed: {
                                 if ( pathv.model.get(index).state != icon_inactive_state) {
-                                    icon_image.width = 50
-                                    icon_image.height = 50
+                                    icon_image.width = 50 * root.scalew
+                                    icon_image.height = 50 * root.scaleh
                                 }
                             }
                             onReleased: {
-                                icon_image.width = 64
-                                icon_image.height = 64
+                                icon_image.width = 64 * root.scalew
+                                icon_image.height = 64 * root.scaleh
                             }
 
                             onExited: {
-                                icon_image.width = 64
-                                icon_image.height = 64
+                                icon_image.width = 64 * root.scalew
+                                icon_image.height = 64 * root.scaleh
                                 if ( pathv.model.get(index).state == icon_normal_bright_state)  {
                                     pathv.model.setProperty(index, "icon", pathv.model.get(index).icon_normal)
                                     pathv.model.setProperty(index, "state", icon_normal_state)
@@ -3172,16 +3228,31 @@ Rectangle {
                     }
                 }
                 path: Path {
-                    startX: 110; startY: 15
+                    startX: 110 * root.scalew
+                    startY: 15 * root.scaleh
                     //startX: 130; startY: 50
                     PathAttribute { name: "iconz"; value: 0 }
                     PathAttribute { name: "iconScale"; value: 0.3 }
                     PathAttribute { name: "iconOpacity"; value: 0.9 }
-                    PathArc { x: 100; y:105; radiusX: 80; radiusY: 60; direction: PathArc.Clockwise; useLargeArc : false}
+                    PathArc {
+                        x: 100 * root.scalew
+                        y: 105 * root.scaleh
+                        radiusX: 80 * root.scalew
+                        radiusY: 60 * root.scaleh
+                        direction: PathArc.Clockwise
+                        useLargeArc : false
+                    }
                     PathAttribute { name: "iconz"; value: 50 }
                     PathAttribute { name: "iconScale"; value: 1.0 }
                     PathAttribute { name: "iconOpacity"; value: 1.0 }
-                    PathArc { x: 20; y: 25; radiusX: 50; radiusY: 60; direction: PathArc.Clockwise; useLargeArc : false}
+                    PathArc {
+                        x: 20 * root.scalew
+                        y: 25 * root.scaleh
+                        radiusX: 50 * root.scalew
+                        radiusY: 60 * root.scaleh
+                        direction: PathArc.Clockwise
+                        useLargeArc : false
+                    }
                     PathAttribute { name: "iconScale"; value: 0.3}
                     PathAttribute { name: "iconOpacity"; value: 0.9 }
 
@@ -3195,10 +3266,10 @@ Rectangle {
         Rectangle {
             id: display2_window
             color: "transparent"
-            x: (root.width/(1280/970));
-            y: (root.height/(720/10));
-            width: (root.width/(1280/300));
-            height: (root.height/(720/220))
+            x: 970 * root.scalew;
+            y: 10 * root.scaleh;
+            width: 300 * root.scalew
+            height: 220 * root.scaleh
             property bool reset: false
 
             MouseArea {
@@ -3216,12 +3287,12 @@ Rectangle {
 
                     if (arrow.visible == false) {
                         if (display2_state == camera1_state) {
-                            micro.x = (root.width/(1280/970))
-                            micro.y = (root.height/(720/180))
+                            micro.x = 970 * root.scalew
+                            micro.y = 180 * root.scaleh
                         }
                         if (display1_state == camera1_state) {
-                            micro.x = (root.width)/(1280/10)
-                            micro.y = (root.height/(720/480))
+                            micro.x = 10 * root.scalew
+                            micro.y = 480 * root.scaleh
                         }
                     }
 
@@ -3257,10 +3328,10 @@ Rectangle {
         Rectangle {
             id: display3_window
             color: "transparent"
-            x: (root.width/(1280/970));
-            y: (root.height/(720/250));
-            width: (root.width/(1280/300));
-            height: (root.height/(720/220))
+            x: 970 * root.scalew;
+            y: 250 * root.scaleh;
+            width: 300 * root.scalew
+            height: 220 * root.scaleh
             property bool reset: false
 
             MouseArea {
@@ -3279,12 +3350,12 @@ Rectangle {
 
                     if (arrow.visible == false) {
                         if (display3_state == camera1_state) {
-                            micro.x = (root.width/(1280/970))
-                            micro.y = (root.height/(720/420))
+                            micro.x = 970 * root.scalew
+                            micro.y = 420 * root.scaleh
                         }
                         if (display1_state == camera1_state) {
-                            micro.x = (root.width)/(1280/10)
-                            micro.y = (root.height/(720/480))
+                            micro.x = 10 * root.scalew
+                            micro.y = 480 * root.scaleh
                         }
                     }
 
@@ -3320,10 +3391,10 @@ Rectangle {
         Rectangle {
             id: display4_window
             color: "transparent"
-            x: (root.width/(1280/970));
-            y: (root.height/(720/486));
-            width: (root.width/(1280/300));
-            height: (root.height/(720/220))
+            x: 970 * root.scalew
+            y: 486 * root.scaleh;
+            width: 300 * root.scalew
+            height: 220 * root.scaleh
             property bool reset: false
 
             MouseArea {
@@ -3341,12 +3412,12 @@ Rectangle {
 
                     if (arrow.visible == false) {
                         if (display4_state == camera1_state) {
-                            micro.x = (root.width/(1280/970))
-                            micro.y = (root.height/(720/655))
+                            micro.x = 970 * root.scalew
+                            micro.y = 655 * root.scaleh
                         }
                         if (display1_state == camera1_state) {
-                            micro.x = (root.width)/(1280/10)
-                            micro.y = (root.height/(720/480))
+                            micro.x = 10 * root.scalew
+                            micro.y = 480 * root.scaleh
                         }
                     }
 
@@ -3383,9 +3454,10 @@ Rectangle {
         Rectangle {
             id: mail_input
             color: "transparent"
-            width: 330; height: 70
-            x: (root.width)/(1280/3)
-            y: (root.height)/(720/18)
+            width: 330 * root.scalew
+            height: 70 * root.scaleh
+            x: 3 * root.scalew
+            y: 18 * root.scaleh
             visible: false
 
             MouseArea {
@@ -3394,8 +3466,8 @@ Rectangle {
                 TextField {
                     id: textfield
                     anchors.horizontalCenter: parent.horizontalCenter
-                    width: 300
-                    height: 40
+                    width: 300 * root.scalew
+                    height: 40 * root.scaleh
                     text: ""
                     onAccepted: focus = false
                     Keys.onEscapePressed: undo()
@@ -3435,8 +3507,8 @@ Rectangle {
             vspmFormat: "BGRA"
             encBitrate: 10485760
             numPframes: 0
-            wlsOutWidth: (root.width)/(1280/960)
-            wlsOutHeight: (root.height)/(720/540)
+            wlsOutWidth: 960 * root.scalew
+            wlsOutHeight: 540 * root.scaleh
             wlsPositionX: 0
             wlsPositionY: 0
             wlsMaxLateness: -1
@@ -3458,10 +3530,10 @@ Rectangle {
         H264Streaming {
             id: camera2
             transmit: false//Camera 2: Receiver
-            wlsOutWidth: (root.width)/(1280/320)
-            wlsOutHeight: (root.height)/(720/240)
-            wlsPositionX: 1440
-            wlsPositionY: 5
+            wlsOutWidth: 320 * root.scalew
+            wlsOutHeight: 240 * root.scaleh
+            wlsPositionX: 960 * root.scalew
+            wlsPositionY: 0
             wlsMaxLateness: -1
             wlsQos: false
             udpPort: "1235"
@@ -3477,10 +3549,10 @@ Rectangle {
         H264Streaming {
             id: camera3
             transmit: false//Camera 3: Receiver
-            wlsOutWidth: (root.width)/(1280/320)
-            wlsOutHeight: (root.height)/(720/240)
-            wlsPositionX: 1440
-            wlsPositionY: (root.height)/(720/240)
+            wlsOutWidth: 320 * root.scalew
+            wlsOutHeight: 240 * root.scaleh
+            wlsPositionX: 960  * root.scalew
+            wlsPositionY: 240 * root.scaleh
             wlsMaxLateness: -1
             wlsQos: false
             udpPort: "1236"
@@ -3496,10 +3568,10 @@ Rectangle {
         H264Streaming {
             id: camera4
             transmit: false//Camera 4: Receiver
-            wlsOutWidth: (root.width)/(1280/320)
-            wlsOutHeight: (root.height)/(720/240)
-            wlsPositionX: 1440
-            wlsPositionY: ((root.height)/(720/240))*2
+            wlsOutWidth: 320 * root.scalew
+            wlsOutHeight: 240 * root.scaleh
+            wlsPositionX: 960  * root.scalew
+            wlsPositionY: 240  * root.scaleh *2
             wlsMaxLateness: -1
             wlsQos: false
             udpPort: "1237"
@@ -3583,10 +3655,10 @@ Rectangle {
         /*CPU monitor*/
         CpuMonitor{
             id: cpu
-            height: (root.height)/(720/120)
-            width: (root.width)/(1280/444)
-            x: (root.width)/(1280/19)
-            y: (root.height)/(720/572)
+            height: 120 * root.scaleh
+            width: 444 * root.scalew
+            x: 19 * root.scalew
+            y: 572 * root.scaleh
             //Adjust the gap between each CPU core display
             m_offset: 2
         }
@@ -3594,14 +3666,14 @@ Rectangle {
         /*MEM monitor*/
         MemMonitor {
             id: mem
-            height: (root.height)/(720/120)
-            width: (root.width)/(1280/256)
-            x: (root.width)/(1280/476)
-            y: (root.height)/(720/572)
+            height: 120 * root.scaleh
+            width: 256 * root.scalew
+            x: 476 * root.scalew
+            y: 572 * root.scaleh
             //Adjust the gap between each display
             m_offset: 5
             //Adjust the height of memory usage bar
-            barHeight: (root.height)/(720/110)
+            barHeight: 110 * root.scaleh
             //Adjust the width of the memory history
             graphWidthOffset: 1
         }
@@ -3609,11 +3681,11 @@ Rectangle {
         /* PulseSignal Display */
         PulseSignal {
             id: bus_graphic
-            x: (root.width)/(1280/742)
-            y: (root.height)/(720/586)
+            x: 742 * root.scalew
+            y: 586 * root.scaleh
             z:5
-            width: (root.width)/(1280/195)//190 - 30
-            height: (root.height)/(720/95)
+            width: 195 * root.scalew        //190 - 30
+            height: 95 * root.scaleh
             displaynumber: false
             fontSize: 10
             maxValue: 2000
@@ -3622,8 +3694,8 @@ Rectangle {
         /* CPU Monitor Text */
         Text {
             id: t_cpu
-            x: (root.width)/(1280/30)
-            y: (root.height)/(720/545)
+            x: 30 * root.scalew
+            y: 545 * root.scaleh
             text: "CPU"
             font.family: "Comic Sans MS"
             font.pointSize: 9
@@ -3631,8 +3703,8 @@ Rectangle {
         }
         Text {
             id: t_cpu_usage
-            x: (root.width)/(1280/25)
-            y: t_cpu.y + (root.height)/(720/10)
+            x: 25 * root.scalew
+            y: t_cpu.y + (10 * root.scaleh)
             text: "Usage"
             font.family: "Comic Sans MS"
             font.pointSize: 9
@@ -3641,7 +3713,7 @@ Rectangle {
 
         Text {
             id: t_cpu1
-            x: (root.width)/(1280/150)
+            x: 150 * root.scalew
             y: t_cpu.y
             text: "CPU1"
             font.family: "Comic Sans MS"
@@ -3650,7 +3722,7 @@ Rectangle {
         }
         Text {
             id: t_cpu1_history
-            x: (root.width)/(1280/145)
+            x: 145 * root.scalew
             y: t_cpu_usage.y
             text: "History"
             font.family: "Comic Sans MS"
@@ -3660,7 +3732,7 @@ Rectangle {
 
         Text {
             id: t_cpu2
-            x: (root.width)/(1280/348)
+            x: 348 * root.scalew
             y: t_cpu.y
             text: "CPU2"
             font.family: "Comic Sans MS"
@@ -3670,7 +3742,7 @@ Rectangle {
         }
         Text {
             id: t_cpu2_history
-            x: (root.width)/(1280/343)
+            x: 343 * root.scalew
             y: t_cpu_usage.y
             text: "History"
             font.family: "Comic Sans MS"
@@ -3681,7 +3753,7 @@ Rectangle {
         /* MEM Monitor Text */
         Text {
             id: mem_usage_text
-            x: (root.width)/(1280/480)
+            x: 480 * root.scalew
             y: bus_reading.y
             text: "0.0"
             font.family: "Comic Sans MS"
@@ -3692,7 +3764,7 @@ Rectangle {
 
         Text {
             id: mem_usage_text_unit
-            x: mem_usage_text.x + (root.width)/(1280/24)
+            x: mem_usage_text.x + (24 * root.scalew)
             y: mem_usage_text.y
             text: "GB"
             font.family: "Comic Sans MS"
@@ -3703,7 +3775,7 @@ Rectangle {
 
         Text {
             id: t_mem
-            x: (root.width)/(1280/485)
+            x: 485 * root.scalew
             y: t_cpu.y
             text: "MEM"
             font.family: "Comic Sans MS"
@@ -3712,7 +3784,7 @@ Rectangle {
         }
         Text {
             id: t_mem_usage
-            x: (root.width)/(1280/482)
+            x: 482 * root.scalew
             y: t_cpu_usage.y
             text: "Usage"
             font.family: "Comic Sans MS"
@@ -3722,7 +3794,7 @@ Rectangle {
 
         Text {
             id: t_mem1
-            x: (root.width)/(1280/618)
+            x: 618 * root.scalew
             y: t_cpu.y
             text: "MEM"
             font.family: "Comic Sans MS"
@@ -3731,7 +3803,7 @@ Rectangle {
         }
         Text {
             id: t_mem1_history
-            x: (root.width)/(1280/613)
+            x: 613 * root.scalew
             y: t_cpu_usage.y
             text: "History"
             font.family: "Comic Sans MS"
@@ -3742,7 +3814,7 @@ Rectangle {
         /* PulseSignal Monitor Text */
         Text {
             id: t_bus
-            x: (root.width)/(1280/823)
+            x: 823 * root.scalew
             y: t_cpu.y
             text: "BUS"
             font.family: "Comic Sans MS"
@@ -3751,7 +3823,7 @@ Rectangle {
         }
         Text {
             id: t_bus_usage
-            x: (root.width)/(1280/818)
+            x: 818 * root.scalew
             y: t_cpu_usage.y
             text: "Usage"
             font.family: "Comic Sans MS"
@@ -3761,8 +3833,8 @@ Rectangle {
 
         Text {
             id: bus_monitor_text
-            x: (root.width)/(1280/10)
-            y: (root.height)/(720/480)
+            x: 10 * root.scalew
+            y: 480 * root.scaleh
             text: ""
             font.family: "Comic Sans MS"
             font.pointSize: 9
@@ -3773,8 +3845,8 @@ Rectangle {
         /*Text indicate BUS Read value*/
         Text {
             id: bus_reading
-            x:bus_graphic.x + (root.width)/(1280/10)
-            y:bus_graphic.y + bus_graphic.height + (root.height)/(720/-1)
+            x:bus_graphic.x + (10 * root.scalew)
+            y:bus_graphic.y + bus_graphic.height + ((-1) * root.scaleh)
             z:5
             text: "READ: "
             color: "green"
@@ -3786,7 +3858,7 @@ Rectangle {
         /*Text indicate BUS Read value*/
         Text {
             id: bus_writing
-            x: bus_reading.x + bus_graphic.width/2 + (root.width)/(1280/15)
+            x: bus_reading.x + bus_graphic.width/2 + (15 * root.scalew)
             y: bus_reading.y
             z:5
             text: "WRITE: "
@@ -3854,11 +3926,11 @@ Rectangle {
         /*Control DDR BUS measurement*/
         Rectangle {
             id: ddrBt
-            x: bus_graphic.x - 1
-            y: bus_graphic.y - (root.height)/(720/15)
+            x: bus_graphic.x - (1 * root.scalew)
+            y: bus_graphic.y - (15 * root.scaleh)
             z:6
-            height: (root.height)/(720/11)
-            width: (root.width)/(1280/35)
+            height: 11 * root.scaleh
+            width: 35 * root.scalew
             color: "transparent"
             border.width: 1
             border.color: "#a5efec"
@@ -3894,8 +3966,8 @@ Rectangle {
             x: ddrBt.x + ddrBt.width + 2
             y: ddrBt.y
             z:6
-            height: (root.height)/(720/11)
-            width: (root.width)/(1280/35)
+            height: 11 * root.scaleh
+            width: 35 * root.scalew
             color: "blue"
             border.width: 1
             border.color: "#a5efec"
@@ -3929,11 +4001,11 @@ Rectangle {
         /*Control GPU BUS measurement*/
         Rectangle {
             id: gpuBt
-            x: cpuBt.x + cpuBt.width + 2
+            x: cpuBt.x + cpuBt.width + (2 * root.scalew)
             y: cpuBt.y
             z:6
-            height: (root.height)/(720/11)
-            width: (root.width)/(1280/35)
+            height: 11 * root.scaleh
+            width: 35 * root.scalew
             color: "blue"
             border.width: 1
             border.color: "#a5efec"
@@ -3967,10 +4039,10 @@ Rectangle {
         Rectangle {
             id: border
             x: ddrBt.x
-            y: ddrBt.y + ddrBt.height - 1
+            y: ddrBt.y + ddrBt.height - (1 * root.scaleh)
             z:6
-            width: (root.width)/(1280/195)//190 - 30
-            height: cpu.height - (root.height)/(720/10)
+            width: 195 * root.scalew           //190 - 30
+            height: cpu.height - (10 * root.scaleh)
             color: "transparent"
             border.width: 0.5
             border.color: "#a5efec"
@@ -4057,14 +4129,14 @@ Rectangle {
 
                     // prevent this block is called again
                     timer_counter = 1 + timer_counter
-                }else if(timer_counter == 7){
+
                     // start video streaming
                     camera1.play()
                     camera2.play()
                     camera3.play()
                     camera4.play()
                     timer.stop()
-                }else if(timer_counter < 7){
+                } else if(timer_counter < 5){
                     timer_counter = 1 + timer_counter
                 }
             }
@@ -4072,8 +4144,8 @@ Rectangle {
 
         Text {
             id: time
-            x:  root.width - time.width - (root.width)/(1280/320) - (root.width)/(1280/10)
-            y:  (root.height)/(720/510)
+            x:  root.width - time.width - (320 * root.scalew) - (10 * root.scalew)
+            y:  510 * root.scaleh
             font.family: "Comic Sans MS"
             font.pointSize: 9
             color: "Yellow"
@@ -4082,8 +4154,8 @@ Rectangle {
 
         Text {
             id: streamurl
-            x:  root.width - streamurl.width - (root.width)/(1280/320) - (root.width)/(1280/10)
-            y:  (root.height)/(720/35)
+            x:  root.width - streamurl.width - (320 * root.scalew) - (10 * root.scalew)
+            y:  50 * root.scaleh
             font.family: "Comic Sans MS"
             font.pointSize: 9
             color: "Orange"
@@ -4099,8 +4171,8 @@ Rectangle {
             Image {
                 id: sun
                 source: "pics/Sun.png"
-                x:  (root.width)/(1280/(-100))
-                y:  (root.height)/(720/(-260))
+                x:  -100 * root.scalew
+                y:  -260 * root.scaleh
                 opacity: 0.0
 
                 //scale: 0.2
@@ -4135,12 +4207,14 @@ Rectangle {
         /*List view for file in folder*/
         Rectangle {
             id: listViewRect
-            width: 330
-            height: 250
+            width: 330 * root.scalew
+            height: 250 * root.scaleh
             border.color: "grey"
             border.width: 1
             radius: 20
-            x: (root.width)/(1280/(-330)); y: (root.height)/(720/190); z: 10;
+            x: (-330) * root.scalew
+            y: 190 * root.scaleh
+            z: 10;
             visible: false
             gradient: Gradient {
                 GradientStop { position: 0.0; color: "grey" }
@@ -4155,7 +4229,7 @@ Rectangle {
                 },
                 State {
                     name: "stage2"; when: isShowRecList == false
-                    PropertyChanges { target: listViewRect; x: -330; visible: false}
+                    PropertyChanges { target: listViewRect; x: ((-330) * root.scalew); visible: false}
                 }
             ]
 
@@ -4174,9 +4248,10 @@ Rectangle {
 
             ListView {
                 id: listView
-                width: 315
-                height: 185
-                x: (root.width)/(1280/10); y: (root.height)/(720/20);
+                width: 315 * root.scalew
+                height: 185 * root.scaleh
+                x: 10 * root.scalew
+                y: 20 * root.scaleh
                 highlight: Rectangle { color: "red"; radius: 3 ; opacity: 0.7 }
                 highlightFollowsCurrentItem: true
                 highlightMoveVelocity: 900 //Speed of movement highlightItem
@@ -4197,8 +4272,8 @@ Rectangle {
 
                     Item {
                         id: lwEleItem
-                        width: 315
-                        height: 30
+                        width: 315 * root.scalew
+                        height: 30 * root.scaleh
                         z: 11
                         Text {
                             color: "#FFFFFF"
@@ -4208,7 +4283,7 @@ Rectangle {
                             id: fName
                             color: "#FFFFFF"
                             text: fileName
-                            x: (root.width)/(1280/30)
+                            x: 30 * root.scalew
                         }
 
                         // Highlight selected item
@@ -4235,16 +4310,16 @@ Rectangle {
 
             Rectangle {
                 //id: titleRlRect
-                y: (root.height)/(720/(-10))
-                width: 330
-                height: 30
+                y: (-10) * root.scaleh
+                width: 330 * root.scalew
+                height: 30 * root.scaleh
                 border.color: "grey"
                 border.width: 1
                 radius: 20
 
                 Text {
-                    x: (root.width)/(1280/50)
-                    y: (root.height)/(720/5)
+                    x: 50 * root.scalew
+                    y: 5 * root.scaleh
                     text: "Please slelect recorded file to play"
                 }
 
@@ -4258,9 +4333,9 @@ Rectangle {
             Rectangle {
                 id: cancelRlBt
                 //x: 10
-                y: (root.height)/(720/220)
-                width: 150
-                height: 30
+                y: 220 * root.scaleh
+                width: 150 * root.scalew
+                height: 30 * root.scaleh
                 color: "transparent"
 
                 Image {
@@ -4268,8 +4343,8 @@ Rectangle {
                     anchors.verticalCenter: cancelRlBt.verticalCenter
                     anchors.horizontalCenter: cancelRlBt.horizontalCenter
                     //anchors.horizontalCenter: nameText.horizontalCenter
-                    width: 150
-                    height: 30
+                    width: 150 * root.scalew
+                    height: 30 * root.scaleh
                     source: "pics/button_cancel.png"
                 }
 
@@ -4304,19 +4379,19 @@ Rectangle {
             /*Recording list OK Button*/
             Rectangle {
                 id: okRlBt
-                x: (root.width)/(1280/175)
-                y: (root.height)/(720/220)
+                x: 175 * root.scalew
+                y: 220 * root.scaleh
                 color: "transparent"
-                width: 150
-                height: 30
+                width: 150 * root.scalew
+                height: 30 * root.scaleh
 
                 Image {
                     id: okRlBt_image
                     anchors.verticalCenter: okRlBt.verticalCenter
                     anchors.horizontalCenter: okRlBt.horizontalCenter
                     //anchors.horizontalCenter: nameText.horizontalCenter
-                    width: 150
-                    height: 30
+                    width: 150 * root.scalew
+                    height: 30 * root.scaleh
                     source: "pics/button_ok.png"
                 }
 
@@ -4398,9 +4473,10 @@ Rectangle {
         Rectangle {
             id: face_detect
             color: "transparent"
-            width: 60; height: 60
-            x: (root.width)/(1280/678)
-            y: (root.height)/(720/10)
+            width: 60 * root.scalew
+            height: 60 * root.scaleh
+            x: 678 * root.scalew
+            y: 10 * root.scaleh
             Image{
                 id: face_icon
                 source: "pics/facedetect.png"
@@ -4531,17 +4607,17 @@ Rectangle {
         }
         if (arrow.visible == false) {
             if (display1_state == camera1_state) {
-                micro.x = (root.width)/(1280/10)
-                micro.y = (root.height/(720/480))
+                micro.x = 10 * root.scalew
+                micro.y = 480 * root.scaleh
             }else if (display2_state == camera1_state) {
-                micro.x = (root.width/(1280/970))
-                micro.y = (root.height/(720/180))
+                micro.x = 970 * root.scalew
+                micro.y = 180 * root.scaleh
             }else if (display3_state == camera1_state) {
-                micro.x = (root.width/(1280/970))
-                micro.y = (root.height/(720/420))
+                micro.x = 970 * root.scalew
+                micro.y = 420 * root.scaleh
             }else if (display4_state == camera1_state) {
-                micro.x = (root.width/(1280/970))
-                micro.y = (root.height/(720/655))
+                micro.x = 970 * root.scalew
+                micro.y = 655 * root.scaleh
             }
         }
         btn_4_3.enabled = false
-- 
1.9.1

